src/storage/test/StorageDAGBenchmark.cpp

```diff
@@ -56,7 +56,7 @@ class FutureDAG {
           // add dependency of root node
           node->addDependency(input.get());
         }
-        if (!node->hasDependents_) {
+        if (!node->isDependent_) {
```

Comment:
Maybe a method return (or direct use) dependencies_.empty() is enough?

Reply:
I don't think it works. As an example, in the tree structure below, what this `if` branch achieves is it appends the `GetNeighborsNode` to `output`. If we look at the `hasDependents_ ` vector in this tree, you can find that only the leaf nodes have empty vector. All other nodes including the root have non-empty vector. It's impossible to distinguish the root from others via the vector size.
![image](https://user-images.githubusercontent.com/19355821/143928236-32c40473-fc6b-484b-8dde-472e6f73ffb2.png)

Reply:
Can you approve the change if it looks good to you? Thank you!